framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/jobs/IJobManager.java

```diff
@@ -0,0 +1,48 @@
+package org.jboss.tools.rsp.server.spi.jobs;
+
+import org.jboss.tools.rsp.eclipse.core.runtime.IStatus;
+
+public interface IJobManager {
+	/**
+	 * Add a job listener to this model
+	 * @param l
+	 */
+	public void addJobListener(IJobListener l);
+	
+	/**
+	 * Remove a job listener from this model
+	 * @param l
+	 */
+	public void removeJobListener(IJobListener l);
+	
+	/**
+	 * Add a job to this model
+	 * @param job
+	 */
+	public void addJob(IJob job);
+	
+	/**
+	 * Remove a job from this model. 
+	 * The status object should indicate the reason for removal
+	 *    (ie, complete (OK), errored, or canceled
+	 *    
+	 * @param job
+	 * @param status
+	 */
+	public void removeJob(IJob job, IStatus status);
+	
+	/**
+	 * Create a job ID for an extender to use to register a custom job
+	 * @return
+	 */
+	public String generateJobId();
```

Comment:
Couldn't this be set implicity to the job upon addition? #addJob could also return it to the caller.